api/src/test/java/org/pmiops/workbench/api/ProfileControllerTest.java

```diff
@@ -736,6 +734,41 @@ public void updateName_alsoUpdatesDucc() {
     assertThat(dbUser.getDuccAgreement().isUserNameOutOfDate()).isTrue();
   }
 
+  @Test
+  public void testSubmitDUCC_name_no_longer_out_of_date() {
+    createAccountAndDbUserWithAffiliation();
+    Profile profile = profileController.getMe().getBody();
+    String givenName1 = profile.getGivenName();
+    String familyName1 = profile.getFamilyName();
+    String initials1 = "AAA";
+
+    profileController.submitDUCC(CURRENT_DUCC_VERSION, initials1);
+    DbUserCodeOfConductAgreement duccAgreement = dbUser.getDuccAgreement();
+    assertThat(duccAgreement.isUserNameOutOfDate()).isFalse();
+    assertThat(duccAgreement.getUserGivenName()).isEqualTo(givenName1);
+    assertThat(duccAgreement.getUserFamilyName()).isEqualTo(familyName1);
+    assertThat(duccAgreement.getUserInitials()).isEqualTo(initials1);
+    assertThat(duccAgreement.getSignedVersion()).isEqualTo(CURRENT_DUCC_VERSION);
+
+    String givenName2 = profile.getGivenName() + " Jr.";
+    String familyName2 = profile.getFamilyName() + " Jr.";
+    String initials2 = "BBB";
+
+    profile.setGivenName(givenName2);
+    profile.setFamilyName(familyName2);
+    profileController.updateProfile(profile);
+    assertThat(dbUser.getDuccAgreement().isUserNameOutOfDate()).isTrue();
+
+    // signing again updates the name and also the out-of-date flag
```

Comment:
oops - before now, we never unset this flag once it has been set.

Reply:
nice catch!